09_maps/main.go

```diff
@@ -111,18 +122,268 @@ func main() {
+	// debug
+	// fmt.Println(heightAdjust)
+	// fmt.Println(weightConverted)
+	bmi = (weightConverted / (heightConverted * heightConverted))
+	bmi = bmi * 10000
+	// debug
+	// fmt.Println(bmi)
+	// fmt.Println("BMI is", bmi)
+	return bmi
+}
+
+// TODO: Create a function that takes an array of maps as an argument and
+// prints the average BMI of the people in the array.
+
+// create function
+// accept array of maps as input
+// loop through all arrays (e.g. "people" array of maps)
+// calculate BMI (calcBMI function)
+// add the sum of all array elements
+// increase counter
+// divide through the amount of array elements
+// print result = avg BMI
+func calcAvgBMI(arrayOfMapInput []map[string]int) float64 {
+	var counter int
+	var bmiSum float64
+	var avgBMI float64
+	var lenConvert int = len(arrayOfMapInput)
+	var lenConverted float64 = float64(lenConvert)
+	for {
+		if counter >= len(arrayOfMapInput) {
+			break
+		}
+		bmiSum += calcBMI(arrayOfMapInput[counter])
+		// debug
+		// fmt.Println(bmiSum)
+		counter++
+	}
+	// debug
+	// fmt.Println(lenConverted)
+	avgBMI = bmiSum / lenConverted
+	// debug
+	// fmt.Println("avg debug print", avgBMI)
+	return avgBMI
+}
+
+// TODO: Create a function that takes an array of maps as an argument and
+// returns the map with the highest BMI.
+
+// create function
+// accept array of maps as input
+// calculate the BMI with function calcBMI for all elements of the array
+// compare all elements with each other to get highest value/BMI
+// if one element is bigger than ALL of the others, return that element
+// else continue comparing
+
+// helper functions
+// write a function that takes a list of int and return the biggest, or 0.
+func largestInt(s []int) int {
+	// counter
+	var counter int
+	// assign variable largest to element [0] of array
+	var largest int
+	// if array length is 0, return 0
+	if len(s) == 0 {
+		return 0
+	}
+	// loop
+	for {
+		// if counter is higher than length of array, break
+		if counter > len(s)-1 {
+			break
+		}
+		// if int of element of array is higher than largest, reassign that element
+		if s[counter] > largest {
+			largest = s[counter]
+		}
+		// increase counter
+		counter++
+	}
+	// return largest int
+	return largest
+}
+
+// helper functions
+// write a function that takes a list of string and return the longest string, or an empty string.
+func largestString(s []string) string {
+	// counter
+	var counter int
+	// initialize largest
+	var largestStringResult string
+	// get length of s[0], assign to variable
+	var largestInitial int = len(s[0])
+	// if array of strings is empty, return empty string
+	if len(s) == 0 {
+		return ""
+	}
+	// loop
+	for {
+		// if counter is higher than the length of array, break
+		if counter > len(s)-1 {
+			break
+		}
+		// if length of element of array is higher than variable largest, re assign that length of value to largest
+		if len(s[counter]) > largestInitial {
+			largestStringResult = s[counter]
+		}
```

Comment:
This is not correct, this compares the length of the current string with the length of the first string in the list.

E.g.
```go
largestString([]string{"a", "abcd", "ab"}) // returns "ab", but should return "abcd"
```

Reply:
Should be fixed.

Reply:
Nop, you are still comparing against the first string in the list.

Reply:
Confused, thought I did the change. But now 🥴

Reply:
The code still isn't correct :grimacing:

Reply:
Now?